packages/components/src/components/hds/disclosure-primitive/index.ts

```diff
@@ -33,11 +33,38 @@ export interface HdsDisclosurePrimitiveSignature {
 }
 
 export default class HdsDisclosurePrimitiveComponent extends Component<HdsDisclosurePrimitiveSignature> {
-  @tracked isOpen = this.args.isOpen ?? false;
+  @tracked _isOpen = false;
+  @tracked _isControlled = this.args.isOpen != undefined;
```

Comment:
when I implemented the other controlled/uncontrolled components I remember it was not possible to 100% determine the internal and external states at the same time, using this "trick": at times they would go out of sync (and I know that in the `Tabs` is implemented this way.

see this video for example
https://github.com/hashicorp/design-system/assets/686239/e46f6a31-c2ef-4c65-9abc-84547c26145c

if you look at this code for example: https://github.com/prysmex/ember-eui/blob/master/packages/core/src/components/eui-accordion.gts you will see how they have two distinct variables, `isInitiallyOpen` and `isOpen`, I suspect exactly for this reason.

happy to chat more and see if I can help with my (fainted) memory of how I implemented the code time ago (or maybe @zamoore has better knowledge/ideas)

Reply:
looking at the referenced code, conceptually `@initialIsOpen` is our `@isOpen` (avoided changing it as it would be a breaking change) and their `_opened` is our `_isOpen`. thanks for sharing that video, I seem to have missed a usacase – I'll look a bit more to understand how they get out of sync.